src_c/simd_blitters_avx2.c

```diff
@@ -582,3 +582,321 @@ blit_blend_rgb_sub_avx2(SDL_BlitInfo *info)
 }
 #endif /* defined(__AVX2__) && defined(HAVE_IMMINTRIN_H) && \
           !defined(SDL_DISABLE_IMMINTRIN_H) */
+
+#if defined(__AVX2__) && defined(HAVE_IMMINTRIN_H) && \
+    !defined(SDL_DISABLE_IMMINTRIN_H)
+void
+blit_blend_rgba_max_avx2(SDL_BlitInfo *info)
+{
+    int n;
+    int width = info->width;
+    int height = info->height;
+
+    Uint32 *srcp = (Uint32 *)info->s_pixels;
+    int srcskip = info->s_skip >> 2;
+    int srcpxskip = info->s_pxskip >> 2;
+    int srceightpxskip = 8 * srcpxskip;
+
+    Uint32 *dstp = (Uint32 *)info->d_pixels;
+    int dstskip = info->d_skip >> 2;
+    int dstpxskip = info->d_pxskip >> 2;
+    int dsteightpxskip = 8 * dstpxskip;
+
+    int pre_8_width = width % 8;
+    int post_8_width = (width - pre_8_width) / 8;
+
+    __m256i *srcp256 = (__m256i *)info->s_pixels;
+    __m256i *dstp256 = (__m256i *)info->d_pixels;
+
+    __m128i mm_src, mm_dst;
+    __m256i mm256_src, mm256_dst;
+
+    while (height--) {
+        if (pre_8_width > 0) {
+            LOOP_UNROLLED4(
+                {
+                    mm_src = _mm_cvtsi32_si128(*srcp);
+                    mm_dst = _mm_cvtsi32_si128(*dstp);
+
+                    mm_dst = _mm_max_epu8(mm_dst, mm_src);
+
+                    *dstp = _mm_cvtsi128_si32(mm_dst);
+
+                    srcp += srcpxskip;
+                    dstp += dstpxskip;
+                },
+                n, pre_8_width);
+        }
+        srcp256 = (__m256i *)srcp;
+        dstp256 = (__m256i *)dstp;
+        if (post_8_width > 0) {
+            LOOP_UNROLLED4(
+                {
+                    mm256_src = _mm256_loadu_si256(srcp256);
+                    mm256_dst = _mm256_loadu_si256(dstp256);
+
+                    mm256_dst = _mm256_max_epu8(mm256_dst, mm256_src);
+
+                    _mm256_storeu_si256(dstp256, mm256_dst);
+
+                    srcp256++;
+                    dstp256++;
+                    srcp += srceightpxskip;
+                    dstp += dsteightpxskip;
+                },
+                n, post_8_width);
+        }
+        srcp += srcskip;
+        dstp += dstskip;
+    }
+}
+#else
+void
+blit_blend_rgba_max_avx2(SDL_BlitInfo *info)
+{
```

Comment:
Just thinking... should places like this raise an exception? I believe this function is defined as an empty function when avx is not available at compile time, but this gets called when avx is available at runtime? In this case, no blitting would happen at runtime and there'd be no warning/error too?

Another alternative is to use the non-SIMD path in this case, and warn something like this at runtime `"your CPU supports AVX2 but pygame was not compiled to use it. Falling back to slower non-SIMD codepath."`

Reply:
All wheels pygame ships should have avx2 support... but maybe some linux distro packagers would not enable it at compile time?

Reply:
Hmm. I'll have a quick think about it. 

I do want to be careful not to accidentally spread avx2 optimizations beyond this .c file, as I know that is an issue that even google managed to mess up, but I think it should be OK in this case as we are in normal C land to call another function from the simd_blitters.h file. Right now it is not possible to compile pygame without SSE2 support so we don't really need to worry further than that...

Yeah I think it should work.

Reply:
Nice, this seems like it should work now! 
But maybe this should raise a warning too before falling back to sse2?

Also, theoretically there's a possibility that the sse2 fallback is called on a CPU that does not support it, but given that this block is only executed after checking for avx2 ability, this should not cause any problems practically because I imagine no CPU that supports avx2 would unsupport sse2